pkg/admissionreview/handler.go

```diff
@@ -89,14 +90,22 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		responseAdmissionReview.Response = &v1.AdmissionResponse{
 			UID:     requestedAdmissionReview.Request.UID,
 			Allowed: funcErr == nil,
-			Result: &metav1.Status{
-				Message: func() string {
-					if funcErr == nil {
-						return ""
-					}
-					return funcErr.Error()
-				}(),
-			},
+			Result: func() *metav1.Status {
+				s, ok := funcErr.(errors.APIStatus)
+				if ok {
+					sts := s.Status()
```

Comment:
guess this should check nil first

Reply:
I'm not sure if you're referring to s or sts, but i think that if s was nil, then ok would be false, and s.Status() can't return nil as its return type is metav1.Status, not *metav1.Status.

Reply:
`funcErr` or `s` could be nil

Reply:
I believe it wouldn't enter the block in either case: https://play.golang.org/p/5aD5HvZ5gyQ

Reply:
https://play.golang.org/p/Z_4vh29ZrEJ

Reply:
Do we really care about explicitly assigning (*T)(nil) though? In this case checking for nil would require using reflect, right?

Reply:
I thought you'd just check s after casting but I haven't double checked

Reply:
For the usual cases (untyped nil) checking `ok` is enough, as it checks whether funcErr is not nil and implements the interface. For (*T)(nil) we would have to use reflect, but then it would probably make sense to check other comparisons in this function to `reflect` ones.
There is a painfully long conversation about this [here](https://groups.google.com/g/golang-nuts/c/wnH302gBa4I).
I'd just stick with the type assertion.

Reply:
you're right, we are casting into an interface so the nil check wouldn't work, I think it's fine then and the main nil value is already handled in the cast